genai-perf/tests/test_image_retrieval_profile_data_parser.py

```diff
@@ -138,3 +139,71 @@ def test_image_retrieval_profile_data(
 
         check_image_retrieval_metrics(metrics, expected_metrics)
         check_statistics(statistics, expected_statistics)
+
+    @patch("pathlib.Path.exists", return_value=True)
+    @patch(
+        "builtins.open",
+        new_callable=mock_open,
+        read_data=json.dumps(image_retrieval_profile_data),
+    )
+    @pytest.mark.parametrize(
+        "infer_mode, load_level, expected_metrics",
+        [
+            (
+                "concurrency",
+                "10",
+                {
+                    "request_throughputs": [1 / ns_to_sec(3)],
+                    "request_latencies": [2, 4],
+                    "image_throughputs": [1 / ns_to_sec(1), 3 / ns_to_sec(4)],
+                    "image_latencies": [1, 4 / 3],
+                    "request_goodputs": [1 / ns_to_sec(6)],
+                },
+            ),
+        ],
+    )
+    def test_image_retrieval_profile_data_with_goodput(
+        self,
+        mock_exists,
+        mock_file,
+        infer_mode,
+        load_level,
+        expected_metrics,
+    ) -> None:
+        """Collect image retrieval metrics from profile export data and check values.
+
+        Goodput Constraints
+        * image latency: 1.25e-6 ms
+        * request latency: 2.5e-6 ms
+
+        Metrics
+        * request throughputs
+            - [2 / (7 - 1)] = [1 / ns_to_sec(3)]
+        * request latencies
+            - [3 - 1, 7 - 3] = [2, 4]
+        * image throughputs
+            - [2 / (3 - 1), 3 / (7 - 3)] = [1 / ns_to_sec(1), 3 / ns_to_sec(4)]
+        * image latencies
+            - [(3 - 1) / 2, (7 - 3) / 3] = [1, 4/3]
+        * request goodput
+            - [1 / (7 - 1)] = [1 / ns_to_sec(6)]
+
+        """
+        test_goodput_constraints = {
+            "image_latency": 1.25e-6,  # ms
+            "request_latency": 2.5e-6,  # ms
```

Comment:
is this msec? or sec?

Reply:
msec, the testing data is in ns unit.
`1.25e-6` will further be 1.25ns to be able to compare with `[1, 4/3]`